### ISSUE TITLE
Polygon area

### ISSUE THREAD

**Issue by minchemo** [RafaelBarbosatec/bonfire#358]
My game will contains multiple areas such as desert, grassland and I would like to add an alert text when the player enters a certain area or in visible.

So I add an `Object layer` and containing a `polygon` on the tiled for testing and add it to the game as a completely transparent `GameComponent`.

BonfireWidget
```dart
objectsBuilder: {
            'test_area': (properties) => AreaDecoration(position: properties.position, size: properties.size)
            }),
```

AreaDecoration
```dart
class AreaDecoration extends GameDecoration {
  AreaDecoration({required Vector2 position, required Vector2 size}) : super(position: position, size: size);
}
```

Polygon on Tiled
![1](https://user-images.githubusercontent.com/22622775/208733910-1118cd4a-15ae-4685-be12-f7a071617aaa.PNG)

But I can't get any element while calling this, any advice?
```dart
Iterable<AreaDecoration> visibleArea = gameRef.visibleComponentsByType<AreaDecoration>();
```

**Comment by RafaelBarbosatec** [comment]
In your `AreaDecoration` you can adds the mixin `ObjectCollision` or `Sensor` to get when any other components give contact with this `AreaDecoration`.

```dart
class AreaDecoration extends GameDecoration with Sensor{
  AreaDecoration({required Vector2 position, required Vector2 size}) : super(position: position, size: size);
  
  @override
  void onContact(GameComponent component) {
    /// Here you can do something with the component that did contact 
  }
}
```

To get the object form (polygon), the `properties` provided by `objectsBuilder` there is not a propertie about the object form. I will adds this in the new release, it’s will named `area` and you can use it to configure the `Sensor` or `ObjectCollision`.

**Comment by minchemo** [comment]
> Hi @minchemo !
> 
> In your `AreaDecoration` you can adds the mixin `ObjectCollision` or `Sensor` to get when any other components give contact with this `AreaDecoration`.
> 
> ```dart
> class AreaDecoration extends GameDecoration with Sensor{
>   AreaDecoration({required Vector2 position, required Vector2 size}) : super(position: position, size: size);
>   
>   @override
>   void onContact(GameComponent component) {
>     /// Here you can do something with the component that did contact 
>   }
> }
> ```
> 
> To get the object form (polygon), the `properties` provided by `objectsBuilder` there is not a propertie about the object form. I will adds this in the new release, it’s will named `area` and you can use it to configure the `Sensor` or `ObjectCollision`.

Thanks for answering, the `area` you ready to add which is a properties in Tiled or anything should I configure when add a polygon in `Object layer` ?

There's a polygon coordinates in my map json, is it possible add to game soI can get this polygon's each points coord in game ?  
```json
// test_area object layer
        {
         "draworder":"topdown",
         "id":8,
         "name":"test_area",
         "objects":[
                {
                 "class":"",
                 "height":0,
                 "id":4,
                 "name":"",
                 "polygon":[
                        {
                         "x":0,
                         "y":0
                        }, 
                        {
                         "x":-16.3636363636364,
                         "y":24
                        }, 
                        {
                         "x":15.8181818181818,
                         "y":23.8181818181818
                        }],
                 "rotation":0,
                 "visible":true,
                 "width":0,
                 "x":80.1818181818182,
                 "y":80
                }],
         "opacity":1,
         "type":"objectgroup",
         "visible":true,
         "x":0,
         "y":0
        }
```

**Comment by RafaelBarbosatec** [comment]
I gonna adds the 'area' param in 'properties' in the next release.

**Comment by RafaelBarbosatec** [comment]
Available in version `2.11.10`.

Example:

<img width="396" alt="image" src="https://user-images.githubusercontent.com/10039275/208909108-1923a7af-1c70-4a32-9612-1369b8901b36.png">


```dart
objectsBuilder: {
          'teste': (properties) => Polygon(properties),
},
```

```dart
class Polygon extends GameDecoration with Sensor {
  @override
  final TiledObjectProperties props;

  Polygon(this.props) : super(position: props.position, size: props.size) {
    setupSensorArea(areaSensor: [props.area]);
  }

  @override
  void onContact(GameComponent component) {
    // TODO: implement onContact
    print(component);
  }
}

```

**Comment by minchemo** [comment]
> Available in version `2.11.10`.
> 
> Example:
> 
> <img alt="image" width="396" src="https://user-images.githubusercontent.com/10039275/208909108-1923a7af-1c70-4a32-9612-1369b8901b36.png">
> 
> ```dart
> objectsBuilder: {
>           'teste': (properties) => Polygon(properties),
> },
> ```
> 
> ```dart
> class Polygon extends GameDecoration with Sensor {
>   
>   final TiledObjectProperties props;
> 
>   Polygon(this.props) : super(position: props.position, size: props.size) {
>     setupSensorArea(areaSensor: [props.area]);
>   }
> 
>   @override
>   void onContact(GameComponent component) {
>     // TODO: implement onContact
>     print(component);
>   }
> }
> ```

Thanks for update so fast!
I just upgrade to 2.11.10, but my polygon doesnt work.
Anything did I missed ?

```dart
map: WorldMapByTiled('map/main.tmj', 
          forceTileSize: Vector2(gameController.tileSize, gameController.tileSize), 
          onError: (value) {
            print('error: ${value.toString()}');
          }, 
          objectsBuilder: {
            'test_area': (properties) => AreaDecoration(properties)
          })
```

```dart
import 'dart:developer';

import 'package:bonfire/bonfire.dart';

class AreaDecoration extends GameDecoration with Sensor {
  final TiledObjectProperties props;

  AreaDecoration(this.props) : super(position: props.position, size: props.size) {
    setupSensorArea(areaSensor: [props.area]);
  }
  
  @override
  void onMount() {
    inspect(props); // no inspect
    super.onMount();
  }

  @override
  void onContact(GameComponent component) {
    print(component); // no print while player contact.
  }
}
```
![2](https://user-images.githubusercontent.com/22622775/208918627-6fa6ba56-27fc-4599-a73a-44872114e2e1.PNG)
